src/main/java/com/ht/project/realtimedeliverymarket/category/service/CategoryService.java

```diff
@@ -0,0 +1,27 @@
+package com.ht.project.realtimedeliverymarket.category.service;
+
+import com.ht.project.realtimedeliverymarket.category.model.entity.Category;
+import com.ht.project.realtimedeliverymarket.category.repository.CategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryService {
+
+  @Autowired
+  private CategoryRepository categoryRepository;
+
+  public void addMainCategory(String name) {
+
+    categoryRepository.findByName(name)
+            .ifPresent(s -> {throw new IllegalArgumentException();});
```

Comment:
띄어쓰기를 해주면 더 깔끔할 것 같아요~
```suggestion
            .ifPresent(s -> { throw new IllegalArgumentException(); });
```

Reply:
그리고 이런 경우는 `IllegalArgumentException`을 상속받은 더 구체적인 타입의 예외를 던져주면 예외를 받는 입장에서는 더 명확할 것 같아요